### ISSUE TITLE
Feature Request: Basic RSA auth backend

### ISSUE THREAD

**Issue by uepoch** [hashicorp/vault#5024]
**Intro:**
I'm pretty sure this has been suggested before, since it seems a basic level auth backend.
Please let me know if I missed something

**Is your feature request related to a problem? Please describe.**
Our applications runs on *many* differents platforms, virtualized, bare-metal ,cloud, mesos,..
We use identity backend to map multiple aliases already
And sometimes, in some harsh environments, it's hard to not just use the plain, deployed rsa private key (without going with userpass)

**Describe the solution you'd like**
Be able to have a userpass complexity-level backend, where you can map 1-1 a public key with a private one and identify the entity (and give them policies, optionally)

**Describe alternatives you've considered**
Using approle with fully hardcoded values (role_id, secret_id)
Using userpass
Both of them require that we store the "password" in vault, we would like to avoid it
Also, to have one "dirty" approle, and a classic one, for an application, we would have to multiply approle for the same identity

**Explain any additional use-cases**
I feel this is a basic need for people administrating infrastructures with legacy parts.
This allow for migration from contexts where you can't directly change the authentication informations available
This can totally get implemented as a plugin, but it's generic enough to have its place next to userpass backend.

**Additional context**
Would be happy to implement this if accepted

**Comment by jefferai** [comment]
I don't believe anyone has suggested this before. I think the main problem for most people is that they don't have anywhere to store the private keys securely; and if they did, they could just as easily store a different credential that could be used to authenticate.

**Comment by uepoch** [comment]
I agree with you on second point.
We aim at reducing the "cost" for internal clients (e.g product devs) to integrate with Vault, (through the identity backend, mostly) and for product/technical/... reasons, they may not be able to change what credentials they have available

From our specific pov, on our infra, this would ease greatly migrating applications to vault, independently from the platform they're deployed on.

I feel this is something that could be used by many people relying on identity backend, for low-cost integration using a feeding job from another data source

Would you find it OK to accept a patch upstream or is it too specific ?

**Comment by aphorise** [comment]
Hey @uepoch I was curious where you got to with this?
Is it still relevant to you?
